File: 14.py
# 14. Longest Common Prefix
# Write a function to find the longest common prefix string amongst an array of strings.
# If there is no common prefix, return an empty string "".

def longestCommonPrefix(strs):
    for i in range(len(strs[0])-1, -2, -1):
        prefix = strs[0][0:i+1]
        print(f"prefix: {prefix}")
        count = 0
        for j in range(0, len(strs)):
            if strs[j][0:i+1] == prefix:
                prefix = strs[j][0:i+1]
                count += 1
            if count == len(strs):
                return prefix

print(longestCommonPrefix(["a"]))
print(longestCommonPrefix(["dog","acecar","dcar"]))

File: 14.js
// 14. Longest Common Prefix
// Write a function to find the longest common prefix string amongst an array of strings.
// If there is no common prefix, return an empty string "".

var longestCommonPrefix = function(strs) {
    let prefix = ''
    for (let i = 0; i < strs[0].length; i++) {
        let count = 0
        for (let j = 0; j < strs.length; j++) {
            if (strs[0][i] === strs[j][i]) {
                count += 1
            } else {
                return prefix
            }
            if (count === strs.length) {
                prefix += strs[0][i]
            }
        }
    }
    return prefix
};

console.log(longestCommonPrefix(["a", "b", 'C']))